src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,622 @@
+						course = {
+							teacher: entry.te[0].longname,
+							subject: entry.su[0].longname,
+							classes: classNames,
+							timetable: []
+						};
+						intermediateData.courses.push(course);
+					}
+					
+					// Update timetable of course
+					course.timetable.push({
+						date: entry.date,
+						startTime: entry.startTime,
+						endTime: entry.endTime,
+						room: currentRoomName,
+					});
+
+					// Update first and last date of course
+					if (course.startDate === undefined || course.startDate > entry.date) {
+						course.startDate = entry.date;
+					}
+					if (course.endDate === undefined || course.endDate < entry.date) {
+						course.endDate = entry.date;
+					}
+				}
+			}
+		} else {
+			throw new NotImplemented(`Iteration over WebUntis data must be teachers or rooms.`);
+		}
+
+		const result = [];
+
+		// Collect times for each course
+		// TODO: filter time slots for non-existing Schul-Cloud classes?
+		for (let course of intermediateData.courses) {
+			const times = [];
+			for (const timetableEntry of course.timetable) {
+				const newEntry = {
+					weekday: this.getWeekDay(timetableEntry.date),
+					startTime: this.getStartTime(timetableEntry.startTime),
+					duration: this.getDuration(timetableEntry.startTime, timetableEntry.endTime),
+					room: timetableEntry.room,
+					count: 1,
+				};
+				
+				let entryFound = false;
+				for (const givenEntry of times) {
+					if (givenEntry.weekday === newEntry.weekday
+						&& givenEntry.startTime === newEntry.startTime
+						&& givenEntry.duration === newEntry.duration
+						&& givenEntry.room === newEntry.room) {
+						givenEntry.count += 1;
+						entryFound = true;
+					}
+				}
+				
+				if (!entryFound) {
+					times.push(newEntry);
+				}
+			}
+
+			// Assumption: After 2 entries it is recurring
+			const filteredTimes = times.filter(entry => entry.count >= 2);
+
+			result.push({
+				teacher: course.teacher,
+				classes: course.classes,
+				subject: course.subject,
+				times: filteredTimes,
+				state: 'new'
+			});
+		}
+
+		return result;
+	}
+
+	/**
+	* 
+	* @param {*} metaData 
+	*/
+	async fetchData(metaData) {
+		// Not required as all information are fetched during phase 1
+
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} metadata 
+	*/
+	async storeMetadata(metaData, params) {
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		await Promise.all(metaData.map(async (entry) => {
+			const metadataResults = await webUntisMetadataService.find({ query: {
+				datasourceId: params.datasourceId,
+				teacher: entry.teacher,
+				classes: entry.classes,
+				subject: entry.subject
+			}, paginate: false });
+			const result = metadataResults[0];
+
+			if (result !== undefined) { // patch/replace existing
+				await webUntisMetadataService.patch(result._id, { state: entry.state, times: entry.times });
+			} else { // create new
+				await webUntisMetadataService.create(Object.assign(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use an object spread instead of `Object.assign` eg: `{ ...foo }`.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)